packages/arb-avm-cpp/data_storage/src/arbcore.cpp

```diff
@@ -888,11 +891,26 @@ void ArbCore::operator()() {
             }
         }
 
-        if (save_checkpoint) {
-            ReadWriteTransaction tx(data_storage);
-            save_checkpoint_status = saveCheckpoint(tx);
-            tx.commit();
-            save_checkpoint = false;
+        if (update_cleanup) {
+            const std::lock_guard<std::mutex> lock(cleanup_mutex);
+            checkpoints_min_message_index = checkpoints_min_message_index_input;
+            update_cleanup = false;
+        }
+
+        if (!machineIdle() && checkpoints_min_message_index > 0) {
+            auto status = deleteOldCheckpoint(checkpoints_min_message_index);
+            if (!status.ok()) {
+                if (!status.IsNotFound()) {
+                    core_error_string = "error deleting old checkpoints";
+                    machine_error = true;
+                    std::cerr << "ArbCore error: " << core_error_string << ": "
+                              << status.ToString() << "\n";
+                    break;
```

Comment:
Do we really need to stop ArbCore here? I think it's perfectly safe to log an error and keep going.

Reply:
good point.